finetuning.py

```diff
@@ -0,0 +1,145 @@
+import argparse
+import importlib
+import os
+from enum import Enum, auto
+from typing import List
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.optimizers import Adam
+
+from lr_face.data_providers import Triplet, test_data, make_triplets
+from lr_face.losses import TripletLoss
+from lr_face.models import FinetuneModel
+from lr_face.utils import fix_tensorflow_rtx
+
+# Needed to make TensorFlow 2.x work with RTX Nvidia cards.
+fix_tensorflow_rtx()
+
+
+class BaseModel(Enum):
+    """
+    This Enum can be used to define all base model architectures that we
+    currently support, and to retrieve embedding and finetune models. This
+    abstracts away the individual implementations of various models so that
+    there is one unified way of loading models.
+
+    TODO: currently only supports Tensorflow models.
+
+    To load the embedding model for VGGFace for example, you would use:
+
+        `BaseModel.VGGFace.load_embedding_model()`
+
+    Similarly, to load a finetune model, you would use:
+
+        `BaseModel.VGGFace.load_finetune_model()`
+    """
+    VGGFace = auto()
```

Comment:
is hier simpel support voor lowercase in te butsen? nu moet je precies de hoofdletters volgen (licht gedoe)

Reply:
Hmm goed idee. Ga ik naar kijken.

- [ ] case-insensitive enum members